go/libraries/doltcore/sqle/writer/prolly_index_writer.go

```diff
@@ -267,13 +279,16 @@ func (m prollySecondaryIndexWriter) Map(ctx context.Context) (prolly.Map, error)
 	return m.mut.Map(ctx)
 }
 
-func (m prollySecondaryIndexWriter) Insert(ctx context.Context, sqlRow sql.Row) error {
+func (m prollySecondaryIndexWriter) ValidateKeyViolations(ctx context.Context, sqlRow sql.Row) error {
 	if m.unique {
 		if err := m.checkForUniqueKeyErr(ctx, sqlRow); err != nil {
 			return err
 		}
```

Comment:
might as well inline this method if it's only called here

Reply:
`checkForUniqueKeyErr` is also called in `prollySecondaryIndexWriter.Update()`